pkg/timestream/datasource.go

```diff
@@ -170,6 +170,26 @@ func sliceFromRows(rows []*timestreamquery.Row, doubleQuotes bool) []string {
 	return res
 }
 
+// The dimensions of a row are the diffenrent columns (apart from the measure)
+// It's encoded in every row, as an array of values.
+func dimensionsFromRows(rows []*timestreamquery.Row, doubleQuotes bool) []string {
```

Comment:
I find this much nesting and reassignment difficult to read at a glance. 
I don't know if it reducing it a lot would be clearer. What do you think? 
```suggestion
func dimensionsFromRows(rows []*timestreamquery.Row, doubleQuotes bool) []string {
	res := []string{}
	if len(rows) > 0 && len(rows[0].Data) == 3 && len(rows[0].Data[2].ArrayValue) > 0 {
		for _, dim := range rows[0].Data[2].ArrayValue {
			if len(dim.RowValue.Data) > 0 && dim.RowValue.Data[0].ScalarValue != nil {
				res = append(res, *dim.RowValue.Data[0].ScalarValue)
			}
		}
	}
```